db/channel_cache_single.go

```diff
@@ -563,6 +563,7 @@ func (c *singleChannelCache) prependChanges(changes LogEntries, changesValidFrom
 		copy(c.logs, changes)
 		base.Infof(base.KeyCache, "  Initialized cache of %q with %d entries from query (#%d--#%d)",
 			base.UD(c.channelName), len(changes), changes[0].Sequence, changes[len(changes)-1].Sequence)
+		c.statsMap.Add(base.StatKeyChannelCacheRevsActive, int64(len(changes)))
```

Comment:
```suggestion
	for _, change := range changes {
		c.cachedDocIDs[change.DocID] = struct{}{}
		c.UpdateCacheUtilization(change, 1)
	}
```

Reply:
To minimize the number of iterations through the set, group these together and remove the call to c.addDocIDs (and remove the unused addDocIDs helper altogether).